### ISSUE TITLE
Is Isolation domain same as concurrency domain?

### ISSUE THREAD

**Issue by stzn** [apple/swift-migration-guide#130]
The term “isolation domain” is used to refer to an independent unit of isolation in Data Race Safety.
[Swift 6 Concurrency Migration Guide](https://www.swift.org/migration/documentation/swift-6-concurrency-migration-guide/dataracesafety#Isolation-Domains)

However, it seems that the term “concurrency domain” is used for the same concept in The Swift Programming Language (TSPL).
[The Swift Programming Language - Concurrency](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/concurrency/#Sendable-Types)

If they refer to the same concept, which term is correct?
If these terms are different, what is the difference between them?

**Comment by stzn** [comment]
And "concurrency context" as well.
https://developer.apple.com/documentation/swift/sendable

**Comment by mattmassicotte** [comment]
This is a really great observation. All these terms are equivalent.

@hborla do you have any opinions on which would be the most resonable to standardize around?

**Comment by stzn** [comment]
A similar discussion appears to be presented.
https://github.com/swiftlang/swift/pull/79509#discussion_r1966596269

**Comment by hborla** [comment]
Yeah, originally I was using "isolation domain", but I agree with Sima's comments in the other discussion that "concurrency domain" is probably more approachable, which is why I used that term in the educational notes. I'm leaning toward standardizing on "concurrency domain" and updating the migration guide to use that term instead of "isolation domain". Thoughts?

**Comment by stzn** [comment]
While "concurrency domain" is a valid term, "isolation domain" might be more intuitive since Swift already uses "isolation" terminology like "actor isolation" and "isolated/nonisolated". Understanding the concept of "isolation" could help people better grasp the meaning of "isolation domain" over time.

I'm curious about why "concurrency domain" is considered a more approachable or readable term in the comment in the other discussion. Will "concurrency domain" provide a better entry point for learning concurrency?

**Comment by FranzBusch** [comment]
I personally also prefer `isolation domains` since as @stzn said isolation is spread everywhere in the language e.g. `isolated` parameters.

**Comment by heckj** [comment]
While "concurrency domain" does a good job implying there's a domain's that concurrency resides in, `isolation domain` has the benefit of reflecting the keywords that all use isolation, which gives it a solid link. I personally started mapping that back to the early async/await + actors talk at WWDC a few years back that used islands as a metaphor, which also made me think more of "isolation domain".

When it comes right down to it, if we refit to be consistent I don't think either term will be bad. Just feels like a smidge more of a lift to map "isolated and nonisolated" into that phrasing.

**Comment by hborla** [comment]
I'd also love to hear @amartini51 's thoughts
